Move xofs

charjr opened the pull request:
Merge #202 first

charjr committed:
Move AllOf, AnyOf and OneOf

carnage commented:
Currently will break BC;

Add back in the openAPI/Xofs and have them extend the ones moved to Processor. Mark them as deprecated, we'll remove when we do a major version bump.

charjr commented:
@carnage Created deprecated extensions of each class.

Also moved the docs for them to the general-use processors.

(Also removed docs for a JSON processor which we do not have)
